### DISCUSSION TITLE
c3 bindings generator

### DISCUSSION THREAD

**Discussion by nexovec** [c3lang/c3c#1739] [Q&A]
Hi, to easily access C libraries, I'd like something that generates bindings for it, does anything of this sort exist?

**Comment by lerno** [comment]
There is nothing official, but people have had great success just using regex(!). Note that to access a C library you need to do very little. Say you want `int puts(const char*)` from C3, then you only need to declare a `external int puts(char*);` and you're fine.

**Comment by cce-dsantos** [comment]
apologies to unearth this @lerno, but is there there any discussion in place to address c3c compiling C files straight away?
if so it should be done by simple "import" or "include" a C file as-is and c3c would do its magic to figure out ".c3i", a bit like dland does but way better zig's indeed

**Reply by lerno** [reply]
You mean translating header files? I understand the convenience, but it's rather difficult to do without a full blown C compiler, and even with that the conversion is often subpar, so that actual manually created .c3i files are way superior. There is a bindings generator which can help generate C3 headers though: https://github.com/vssukharev/bc3
